core/src/main/scala/dev/profunktor/auth/jwt.scala

```diff
@@ -25,17 +27,52 @@ object jwt {
 
   case class JwtToken(value: String) extends AnyVal
 
-  case class JwtSecretKey(value: String) extends AnyVal
+  object JwtSecretKey {
+    def apply(key: Array[Byte]): JwtSecretKey = new JwtSecretKeyByteArr(key)
+    def apply(key: Array[Char]): JwtSecretKey = new JwtSecretKeyCharArr(key)
+    def apply(key: PrivateKey): JwtSecretKey  = new JwtSecretKeyPK(key)
+  }
+  sealed trait JwtSecretKey {
+    def value: Array[Char]
+  }
+  private class JwtSecretKeyCharArr(val value: Array[Char]) extends JwtSecretKey
+  private class JwtSecretKeyByteArr(bytes: Array[Byte]) extends JwtSecretKey {
+    lazy val value = {
+      val byteBuffer = java.nio.ByteBuffer.wrap(bytes)
+      val charBuffer = StandardCharsets.UTF_8.decode(byteBuffer)
+      val charArray  = new Array[Char](charBuffer.remaining())
+      charBuffer.get(charArray)
+      charArray
+    }
+  }
+  private class JwtSecretKeyPK(key: PrivateKey) extends JwtSecretKey {
+    lazy val value = {
+      val byteBuffer = java.nio.ByteBuffer.wrap(key.getEncoded())
+      val charBuffer = StandardCharsets.UTF_8.decode(byteBuffer)
+      val charArray  = new Array[Char](charBuffer.remaining())
+      charBuffer.get(charArray)
+      charArray
+    }
+  }
 
   sealed trait JwtAuth
   case object JwtNoValidation extends JwtAuth
   case class JwtSymmetricAuth(secretKey: JwtSecretKey, jwtAlgorithms: Seq[JwtHmacAlgorithm]) extends JwtAuth
   case class JwtAsymmetricAuth(publicKey: JwtPublicKey) extends JwtAuth
   object JwtAuth {
     def noValidation: JwtAuth = JwtNoValidation
+    @deprecated(message = "use of string to hold secret keys is deprecated", since = "1.x")
```

Comment:
```suggestion
    @deprecated(message = "use of string to hold secret keys is deprecated", since = "2.x")
```

I think we have to publish this as 2.x, as it is already breaking and in 1.x there is no possibility to use anything but String